lte/gateway/python/magma/subscriberdb/rpc_servicer.py

```diff
@@ -64,6 +66,28 @@ def DeleteSubscriber(self, request, context):
         logging.debug("Delete subscriber rpc for sid: %s", sid)
         self._store.delete_subscriber(sid)
 
+    @return_void
+    def PurgeSubscriber(self, request, context):
+        """
+        Triggers a network initiated detach
+        """
+        self._print_grpc(request)
+        sid = SIDUtils.to_str(request)
+        logging.debug("Purge subscriber rpc for sid: %s", sid)
+        self._store.delete_subscriber(sid)
+
```

Comment:
What if subsequent code fails. We shall not delete from subsciption db if rest of the logic fails.